tests/LoggerTest.php

```
@@ -191,4 +191,61 @@ public function testNestedSpanLoggerLinksToParentSpanId(): void
         $this->assertNotNull($innerSpan);
         $this->assertSame($outerSpan->id, $innerSpan->parentId);
     }
+
+    public function testNestedSpanLoggerInheritsParentTraceId(): void
+    {
+        $log = new Logger('svc');
+        $outer = $log->createSpanLogger('outer');
+        $inner = $outer->createSpanLogger('inner');
+
+        $this->assertSame($outer->getSpan()->traceId, $inner->getSpan()->traceId);
```

PHPStan (2.2.1)

[warning] Cannot access property $traceId on Timewave\Logger\Classes\Span|null.


(property.nonObject)